scapy/sendrecv.py

```diff
@@ -291,13 +297,10 @@ def _sndrcv_snd(self):
             elif not self._send_done:
                 self.notans = i
             self._send_done = True
-        # In threaded mode, timeout.
-        if self.threaded and self.timeout is not None and not self.breakout:
-            t = time.monotonic() + self.timeout
-            while time.monotonic() < t:
```

Comment:
And while we're at it, this code is disgusting and has terrible perfs. Should have used a lock or something, so I just did that.